logger/src/lib/middlewares/api-logging.middleware.ts

```diff
@@ -0,0 +1,55 @@
+import { NextFunction, Request, Response } from 'express';
+import { LoggerService } from '../services/logger.service';
+
+const loggerService = new LoggerService();
+
+function captureResponseBody(res: Response, callback: (body: any) => void) {
+  let responseBody: any;
+  const { write, end } = res;
+
+  res.write = (...args: any[]) => {
+    const [chunk = null, encoding = 'utf8', cb] = args;
+    responseBody = chunk;
+    return write.apply(res, [chunk, encoding, cb]);
+  };
+
+  res.end = (...args: any[]) => {
+    const [chunk = null, encoding = 'utf8', cb] = args;
+    if (chunk) responseBody = chunk;
+    return end.apply(res, [chunk, encoding, cb]);
+  };
+
+  res.on('finish', () => callback(responseBody));
+}
+
+export function apiLoggingMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  loggerService.info('[REQUEST]', {
+    method: req.method,
+    path: req.path,
+    headers: req.headers,
+    payload: req.body,
+  });
+
+  captureResponseBody(res, (responseBody) => {
+    loggerService.info('[RESPONSE]', {
+      statusCode: res.statusCode,
+      statusMessage: res.statusMessage,
+      headers: res.getHeaders(),
+      body: responseBody?.toString(),
```

Comment:
This is goo @mlalivio , I think we need to whitelist some properties that we shouldn't log e.g. password, pincode, etc.